usr/src/uts/common/fs/zfs/vdev_disk.c

```diff
@@ -324,10 +325,10 @@ vdev_disk_open(vdev_t *vd, uint64_t *psize, uint64_t *max_psize,
 	if (vd->vdev_devid != NULL) {
 		if (ddi_devid_str_decode(vd->vdev_devid, &dvd->vd_devid,
 		    &dvd->vd_minor) != 0) {
-			vd->vdev_stat.vs_aux = VDEV_AUX_BAD_LABEL;
 			vdev_dbgmsg(vd, "vdev_disk_open: invalid "
 			    "vdev_devid '%s'", vd->vdev_devid);
-			return (SET_ERROR(EINVAL));
+			dvd->vd_devid = NULL;
+			dvd->vd_minor = NULL;
 		}
 	}
 
```

Comment:
we also need this fix (wrong field is checked against NULL):
diff --git a/usr/src/uts/common/fs/zfs/vdev_disk.c b/usr/src/uts/common/fs/zfs/vdev_disk.c
index 93b24fcaf1..1b6592344a 100644
--- a/usr/src/uts/common/fs/zfs/vdev_disk.c
+++ b/usr/src/uts/common/fs/zfs/vdev_disk.c
@@ -363,7 +363,7 @@ vdev_disk_open(vdev_t *vd, uint64_t *psize, uint64_t *max_psize,
                /*
                 * Compare the devid to the stored value.
                 */
-               if (error == 0 && vd->vdev_devid != NULL &&
+               if (error == 0 && vd->vd_devid != NULL &&
                    ldi_get_devid(dvd->vd_lh, &devid) == 0) {
                        if (ddi_devid_compare(devid, dvd->vd_devid) != 0) {
                                error = SET_ERROR(EINVAL);

Reply:
Good spot - just fixed and rebased.

Reply:
There are other checks below that need to have the same modification:
https://github.com/openzfs/openzfs/pull/605/files#diff-9ab1cde67323e116c09c49bd9d6a343aR391
https://github.com/openzfs/openzfs/pull/605/files#diff-9ab1cde67323e116c09c49bd9d6a343aR403

Reply:
Thanks - done.